File: backend/src/main/java/com/aisale/backend/entity/Address.java
package com.aisale.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "addresses")
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String receiverName;

    @Column(nullable = false)
    private String receiverPhone;

    @Column(nullable = false)
    private String province;

    @Column(nullable = false)
    private String city;

    @Column(nullable = false)
    private String district;

    @Column(nullable = false, length = 500)
    private String detailAddress;

    private Boolean isDefault = false;

    @Enumerated(EnumType.STRING)
    private AddressStatus status = AddressStatus.ACTIVE;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public enum AddressStatus {
        ACTIVE, DELETED
    }
}

File: frontend/src/api/address.ts
import request from '@/utils/request'

export interface Address {
  id?: number
  receiverName: string
  receiverPhone: string
  province: string
  city: string
  district: string
  detailAddress: string
  isDefault: boolean
  fullAddress?: string
}

/**
 * 获取用户地址列表
 */
export async function getUserAddresses() {
  const res = await request.get('/user/addresses')
  return res.data
}

/**
 * 获取单个地址详情
 */
export async function getAddressById(id: number) {
  const res = await request.get(`/user/addresses/${id}`)
  return res.data
}

/**
 * 创建新地址
 */
export async function createAddress(data: Address) {
  const res = await request.post('/user/addresses', data)
  return res.data
}

/**
 * 更新地址
 */
export async function updateAddress(id: number, data: Address) {
  const res = await request.put(`/user/addresses/${id}`, data)
  return res.data
}

/**
 * 删除地址
 */
export async function deleteAddress(id: number) {
  const res = await request.delete(`/user/addresses/${id}`)
  return res.data
}

/**
 * 设置默认地址
 */
export async function setDefaultAddress(id: number) {
  const res = await request.put(`/user/addresses/${id}/default`)
  return res.data
}
